src/Compilers/CSharp/Portable/Symbols/FieldSymbol.cs

```diff
@@ -445,6 +446,25 @@ public virtual int TupleElementIndex
         {
             get
             {
+                if (ContainingType.IsTupleType && ContainingType.IsDefinition)
+                {
+                    var i = NamedTypeSymbol.MatchesCanonicalElementName(Name);
+                    int itemsPerType = Math.Min(ContainingType.Arity, NamedTypeSymbol.ValueTupleRestPosition - 1);
+                    if (i > 0 && i <= itemsPerType)
+                    {
+                        WellKnownMember wellKnownMember = NamedTypeSymbol.GetTupleTypeMember(ContainingType.Arity, i);
+
+                        RuntimeMembers.MemberDescriptor relativeDescriptor = WellKnownMembers.GetDescriptor(wellKnownMember);
+                        var found = CSharpCompilation.GetRuntimeMember(ImmutableArray.Create<Symbol>(this), relativeDescriptor, CSharpCompilation.SpecialMembersSignatureComparer.Instance,
+                                                                  accessWithinOpt: null); // force lookup of public members only
```

Comment:
This looks expensive. Can we calculate `TupleElementIndex` at most once? (Perhaps this property should be implemented in derived types where it might make more sense to cache the value.)

Reply:
I didn't want to add state to every source or PE symbol just because it might be a tuple element.
By limiting this to definitions of ValueTuple and doing a cheap name check first, I think this becomes reasonable.

---
In reply to: [425346752](https://github.com/dotnet/roslyn/pull/44231#discussion_r425346752) [](ancestors = 425346752)